helix-term/src/application.rs

```diff
@@ -287,10 +287,10 @@ impl Application {
                             main.map(|t| t.id)
                         });
                         if let Some(id) = main {
-                            select_thread_id(&mut self.editor, id, false).await;
+                            select_thread_id(&mut self.editor, id, true).await;
```

Comment:
See the comment before the block, we specifically don't want to force override (that's why I added the parameter). If `thread_id` is already set that means we're already looking at a stopped thread and we don't want to change focus if some other thread happens to stop during that time (but we can log a message).

https://github.com/emacs-lsp/dap-mode/blob/8ec7e98986ea46e3c36b7ecbdf9a6562c90c7632/dap-mode.el#L942

If we press continue then the `continued` event will call `resume-application` (if the thread id matches, looks like I forgot to do that check): https://github.com/emacs-lsp/dap-mode/blob/8ec7e98986ea46e3c36b7ecbdf9a6562c90c7632/dap-mode.el#L934-L936
Which will clear the currently active thread & active_frame:

https://github.com/emacs-lsp/dap-mode/blob/8ec7e98986ea46e3c36b7ecbdf9a6562c90c7632/dap-mode.el#L670-L671

Looks like we need to split out the `continued` logic and call it on `continue` or `step`